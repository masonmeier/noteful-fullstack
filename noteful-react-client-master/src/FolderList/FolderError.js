// Error Boundary component
// NOTE: MUST use class

import React from 'react';

class FolderError extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			hasError: false
		};
	}

	static getDerivedStateFromError(error) {
		// Update state so the next render will show the fallback UI.
		return { hasError: true };
	}

	componentDidCatch(error, errorInfo) {
		// You can also log the error to an error reporting service
		// logErrorToMyService(error, errorInfo);
	}

	render() {
		if (this.state.hasError) {
			// You can render any custom fallback UI
			return <h2>Could not display this folder.</h2>;
		}
		return this.props.children;
	}
}
export default FolderError;
