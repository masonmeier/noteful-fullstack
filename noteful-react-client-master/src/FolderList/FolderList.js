import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import FolderError from './FolderError';
import NotefulContext from '../NotefulContext';
import config from '../config';

const FolderList = props => {
	const contextType = useContext(NotefulContext);
	const { notes, folders, handleClickDeleteFolder } = contextType;

	let folderId;
	if (config.DATASOURCE === 'postgresql') {
		folderId = parseInt(props.match.params.id_folder);
	} else {
		folderId = props.match.params.id_folder;
	}

	return (
		<>
			<header>
				<>
					<h2>Folders</h2>
					&nbsp;&nbsp;
					<NavLink to={'/add-folder'}>
						<button className="btn-add">+</button>
					</NavLink>
					&nbsp;&nbsp;
					{props.match.params.id_folder !== undefined ? (
						<>
							<NavLink to={`/edit-folder/${props.match.params.id_folder}`}>
								<button className="btn-edit">&#9998;</button>
							</NavLink>
							&nbsp;&nbsp;
							<button
								className="btn-delete"
								onClick={() => handleClickDeleteFolder(folderId, props)}
							>
								-
							</button>
						</>
					) : null}
				</>
			</header>
			<ul>
				{folders.map(folder => (
					<li
						key={folder.id}
						className={folder.id === folderId ? ' active' : null}
					>
						<FolderError>
							<NavLink to={`/folders/${folder.id}`}>
								<span role="img" aria-label="Folder">
									&#x1F4C2;
								</span>
								&nbsp;{folder.name}&nbsp;(
								{notes.filter(note => note.id_folder === folder.id).length})
							</NavLink>
						</FolderError>
					</li>
				))}
			</ul>
		</>
	);
};

export default FolderList;

// to catch bugs
// check that get a folders array that has id and name
// this array is the "folders" variable coming from context
FolderList.propTypes = {
	folders: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string.isRequired,
			name: PropTypes.string.isRequired
		})
	)
};
