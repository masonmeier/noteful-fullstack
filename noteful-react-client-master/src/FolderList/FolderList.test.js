import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FolderList from './FolderList';

describe(`FolderList component`, () => {
	const props = {
		match: {
			params: {
				id_folder: '101'
			}
		},
		folders: [
			{
				id: '101',
				name: 'test-folder-label'
			}
		]
	};

	it('renders a FolderList by default', () => {
		const wrapper = shallow(<FolderList {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders the FolderList given props', () => {
		const wrapper = shallow(<FolderList {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
