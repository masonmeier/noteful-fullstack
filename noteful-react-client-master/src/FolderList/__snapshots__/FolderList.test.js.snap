// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderList component renders a FolderList by default 1`] = `
<Fragment>
  <header>
    <h2>
      Folders
    </h2>
      
    <NavLink
      to="/add-folder"
    >
      <button
        className="btn-add"
      >
        +
      </button>
    </NavLink>
      
    <NavLink
      to="/edit-folder/101"
    >
      <button
        className="btn-edit"
      >
        ✎
      </button>
    </NavLink>
      
    <button
      className="btn-delete"
      onClick={[Function]}
    >
      -
    </button>
  </header>
  <ul />
</Fragment>
`;

exports[`FolderList component renders the FolderList given props 1`] = `
<Fragment>
  <header>
    <h2>
      Folders
    </h2>
      
    <NavLink
      to="/add-folder"
    >
      <button
        className="btn-add"
      >
        +
      </button>
    </NavLink>
      
    <NavLink
      to="/edit-folder/101"
    >
      <button
        className="btn-edit"
      >
        ✎
      </button>
    </NavLink>
      
    <button
      className="btn-delete"
      onClick={[Function]}
    >
      -
    </button>
  </header>
  <ul />
</Fragment>
`;
